devserver_lib/src/lib.rs

```diff
@@ -17,6 +16,26 @@ use std::thread;
 #[cfg(feature = "reload")]
 mod reload;
 
+struct FileProxy {
+    path: String,
+    extension: Option<String>
+}
+impl FileProxy {
+    fn new(path: String, extension: Option<String>) -> Self {
+        let owned_ext = match extension {
+            Some(ext) => Some(ext),
+            None => None,
+        };
+        FileProxy { path: path, extension: owned_ext }
+    }
+    fn blank() -> Self {
+        FileProxy { path: String::from(""), extension: None }
```

Comment:
`String::from("")` and `String::new()` do the same thing.

`String::from("")` might do *slightly* more work behind the scenes, but maybe is slightly clearer. It's a matter of personal preference, but I thought I'd point it out.